src/plugins/dashboard/public/dashboard_listing/dashboard_listing.tsx

```diff
@@ -0,0 +1,264 @@
+  }, [dashboardSessionStorage, goToDashboard, useSessionStorageIntegration]);
+
+  const fetchItems = useCallback(
+    (
+      searchTerm: string,
+      {
+        references,
+        referencesToExclude,
+      }: {
+        references?: SavedObjectsFindOptionsReference[];
+        referencesToExclude?: SavedObjectsFindOptionsReference[];
+      } = {}
+    ) => {
+      const searchStartTime = window.performance.now();
+      return findDashboards
+        .findSavedObjects({
+          search: searchTerm,
+          size: listingLimit,
+          hasReference: references,
+          hasNoReference: referencesToExclude,
+        })
+        .then(({ total, hits }) => {
+          const searchEndTime = window.performance.now();
+          const searchDuration = searchEndTime - searchStartTime;
+          reportPerformanceMetricEvent(pluginServices.getServices().analytics, {
+            eventName: SAVED_OBJECT_LOADED_TIME,
+            duration: searchDuration,
+            meta: {
+              saved_object_type: DASHBOARD_SAVED_OBJECT_TYPE,
+            },
+          });
+          return {
+            total,
+            hits: hits.map(toTableListViewSavedObject),
+          };
+        });
+    },
+    [findDashboards, listingLimit]
+  );
+
+  const deleteItems = useCallback(
+    async (dashboardsToDelete: Array<{ id: string }>) => {
+      try {
+        const deleteStartTime = window.performance.now();
+
+        await Promise.all(
+          dashboardsToDelete.map(({ id }) => {
+            dashboardSessionStorage.clearState(id);
+            return savedObjectsClient.delete(DASHBOARD_SAVED_OBJECT_TYPE, id);
+          })
+        );
+
+        const deleteDuration = window.performance.now() - deleteStartTime;
+        reportPerformanceMetricEvent(pluginServices.getServices().analytics, {
+          eventName: SAVED_OBJECT_DELETE_TIME,
+          duration: deleteDuration,
+          meta: {
+            saved_object_type: DASHBOARD_SAVED_OBJECT_TYPE,
+            total: dashboardsToDelete.length,
+          },
+        });
+      } catch (error) {
+        toasts.addError(error, {
+          title: dashboardListingErrorStrings.getErrorDeletingDashboardToast(),
+        });
+      }
+
+      setUnsavedDashboardIds(dashboardSessionStorage.getDashboardIdsWithUnsavedChanges());
+    },
+    [savedObjectsClient, dashboardSessionStorage, toasts]
+  );
+
+  const editItem = useCallback(
+    ({ id }: { id: string | undefined }) => goToDashboard(id, ViewMode.EDIT),
+    [goToDashboard]
+  );
+  const emptyPrompt = (
+    <DashboardListingEmptyPrompt
+      createItem={createItem}
+      goToDashboard={goToDashboard}
+      unsavedDashboardIds={unsavedDashboardIds}
+      setUnsavedDashboardIds={setUnsavedDashboardIds}
+      useSessionStorageIntegration={useSessionStorageIntegration}
+    />
+  );
+
+  const { getEntityName, getTableListTitle, getEntityNamePlural } = dashboardListingTableStrings;
+
+  return (
+    <I18nProvider>
```

Comment:
The one in this file specifically wraps the portable component so that consumers don't have to, and you're right that the Dashboard router already contains one. This will cause some duplication, but I figured it was better to have two providers at different levels here than to require consumers to wrap the component with one.